package/kernel/linux/modules/other.mk

```diff
@@ -1230,3 +1230,36 @@ define KernelPackage/it87-wdt/description
 endef
 
 $(eval $(call KernelPackage,it87-wdt))
+
+
+define KernelPackage/serdev-core
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Serial device bus core support
+  DEPENDS:=@LINUX_4_19
+  KCONFIG:=CONFIG_SERIAL_DEV_BUS
+ifneq ($(wildcard $(LINUX_DIR)/drivers/tty/serdev/core.ko),)
+  FILES:=$(LINUX_DIR)/drivers/tty/serdev/core.ko
+endif
+endef
+
+define KernelPackage/serdev-core/description
+  Kernel module for core support for devices connected via a serial port
+endef
+
+$(eval $(call KernelPackage,serdev-core))
+
+
+define KernelPackage/serdev-ttyport
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Serial device TTY port controller support
+  DEPENDS:=+kmod-serdev-core @LINUX_4_19
+  KCONFIG:= \
+	CONFIG_SERIAL_DEV_BUS=y \
+	CONFIG_SERIAL_DEV_CTRL_TTYPORT=y
```

Comment:
`QCA7000_UART` depends on ``SERIAL_DEV_BUS`, but not `SERIAL_DEV_CTRL_TTYPORT`, so there is no reason to enable `SERIAL_DEV_CTRL_TTYPORT` and force the module to be built-in. Just add `CONFIG_SERIAL_DEV_CTRL_TTYPORT=n` to the serdev-core package.

Reply:
Ok, then I could drop the serdev-ttyport entirely. I prepare a patch, test again and update the PR then.

Reply:
Since I happened to use SERIAL_DEV myself recently, you actually need SERIAL_DEV_CTRL_TTYPORT if the device is attached to a normal tty.

I assume QCA7000 is only found on specific targets, not a generic thing you attach, so maybe just add SERIAL_DEV_BUS / SERIAL_DEV_CTRL_TTYPORT to the target's config with =y and restrict the uart version to those targets.

This also means you don't need the kmod-serdev anymore.